src/vs/workbench/services/textfile/common/encoding.ts

```diff
@@ -236,9 +236,15 @@ async function guessEncodingByBuffer(buffer: VSBuffer): Promise<string | null> {
 
 	// ensure to limit buffer for guessing due to https://github.com/aadsm/jschardet/issues/53
 	const limitedBuffer = buffer.slice(0, AUTO_ENCODING_GUESS_MAX_BYTES);
-	// override type since jschardet expects Buffer even though can accept Uint8Array
+
+	// before guessing jschardet calls toString('binary') on input if it is a Buffer,
+	// since we are using it inside browser environment as well we do conversion ourselves
+	// https://github.com/aadsm/jschardet/blob/v2.1.1/src/index.js#L36-L40
+	let binaryString = encodeLatin1(limitedBuffer.buffer);
+
+	// override type since jschardet expects Buffer even though can accept string
 	// can be fixed once https://github.com/aadsm/jschardet/pull/58 is merged
-	const jschardetTypingsWorkaround = limitedBuffer.buffer as any;
+	const jschardetTypingsWorkaround = binaryString as any;
```

Comment:
I discovered while writing tests for the browser. Now I am ashamed that I did it wrong in my previous PR 😅 On the bright side it is a validation of the fact that we should have tests running in the browser as well :)